sdk/src/wallet/operations/transaction/high_level/delegation/delay.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    client::{api::PreparedTransactionData, secret::SecretManage},
+    types::block::{
+        context_input::{CommitmentContextInput, ContextInput},
+        output::{AddressUnlockCondition, BasicOutput, DelegationId, DelegationOutputBuilder, MinimumOutputAmount},
+    },
+    wallet::{operations::transaction::TransactionOptions, types::TransactionWithMetadata, Wallet},
+};
+
+impl<S: 'static + SecretManage> Wallet<S>
+where
+    crate::wallet::Error: From<S::Error>,
+    crate::client::Error: From<S::Error>,
+{
+    /// Delay a delegation's claiming. The `reclaim_excess` flag indicates whether excess value over the minimum storage
+    /// requirements will be moved to a basic output that is unlockable by the same address which controls the
+    /// delegation. Otherwise it will be added to a new delegation. In this manner, one can delegate for one epoch
+    /// at a time and never lose out on any rewards.
+    pub async fn delay_delegation_claiming(
+        &self,
+        delegation_id: DelegationId,
+        reclaim_excess: bool,
+    ) -> crate::wallet::Result<TransactionWithMetadata> {
+        let prepared_transaction = self
+            .prepare_delay_delegation_claiming(delegation_id, reclaim_excess)
+            .await?;
+
+        self.sign_and_submit_transaction(prepared_transaction, None, None).await
+    }
+
+    /// Prepare to delay a delegation's claiming. The `reclaim_excess` flag indicates whether excess value
+    /// over the minimum storage requirements will be moved to a basic output that is unlockable by the same address
+    /// which controls the delegation.
+    /// Otherwise it will be added to a new delegation. In this manner, one can delegate for one epoch at a time and
+    /// never lose out on any rewards.
+    pub async fn prepare_delay_delegation_claiming(
+        &self,
+        delegation_id: DelegationId,
+        reclaim_excess: bool,
+    ) -> crate::wallet::Result<PreparedTransactionData> {
+        if let Some(delegation_output) = self.get_delegation_output(delegation_id).await {
```

Comment:
I would prefer to have this and have the else block at the end (L137) removed
```suggestion
        let delegation_output =
            self.get_delegation_output(delegation_id)
                .await
                .ok_or(crate::wallet::Error::DelegationTransitionFailed(format!(
                    "no delegation output found with id {delegation_id}"
                )))?;
```

Reply:
thanks, done